src/docker_clojure/core.clj

```diff
@@ -75,16 +75,28 @@
       :build-dir build-dir
       :dockerfile filename)))
 
-(defn build-image [installer-hashes {:keys [docker-tag base-image] :as variant}]
-  (let [image-tag (str "clojure:" docker-tag)
-        _         (println "Pulling base image" base-image)
-        _         (pull-image base-image)
+(defn build-image
+  [installer-hashes {:keys [docker-tag base-image architectures] :as variant}]
+  (let [image-tag     (str "clojure:" docker-tag)
+        _             (println "Pulling base image" base-image)
+        _             (pull-image base-image)
 
         {:keys [dockerfile build-dir]}
         (generate-dockerfile! installer-hashes variant)
 
-        build-cmd (remove nil? ["docker" "build" "--no-cache" "-t" image-tag
-                                "--load" "-f" dockerfile "."])]
+        host-arch     (let [jvm-arch (System/getProperty "os.arch")]
+                        (if (= "aarch64" jvm-arch)
+                          "arm64v8"
+                          jvm-arch))
+        platform-flag (if (contains? (or architectures
+                                         cfg/default-architectures)
+                                     host-arch)
+                        nil
+                        (str "--platform=linux/" (first architectures)))
+
+        build-cmd     (remove nil? ["docker" "buildx" "build" "--no-cache"
```

Comment:
Speaking out loud for sanity check. This is only applicable when we want to build some images locally using this `-main`. Docker Hub reads from the `Dockerfile` directly and run them with their own commands changing our build-cmd to use `buildx` doesn't affect them.

Reply:
That's correct. It uses the manifest and the Dockerfiles, not our build code.